vyper/parser/stmt.py

```diff
@@ -323,12 +324,13 @@ def call(self):
             return external_call.make_external_call(self.stmt, self.context)
 
     def parse_assert(self):
-        tmp = self.context.in_assertion  # backup value
+
+        tmp_assertion_value = self.context.in_assertion  # backup value
```

Comment:
I think you can fix this by adjusting the `set_in_assertion` method to the following.

```python
@contextlib.contextmanager
def set_in_assertion():
    prev_value = self.in_assertion
    self.in_assertion = True
    yield
    self.in_assertion = prev_value
```

This will allow nested assertion contexts that correctly revert to the original value after each block (only reverting to `False` after the final block exits.  This all assumes this is desired behavior in all cases.

Reply:
Brilliant! Didn't think of that at all.